### PR TITLE
[FIX] Access to localStorage in Livechat

### PR DESCRIPTION
**Pull request by jschirrmacher**
## Proposed changes (including videos or screenshots)

This fixes access to localStorage in LiveChat, if the user has third party coockies disabled. 

## Issue(s)
#15616

### CODE REVIEW

**Comment by murtaza98** [comment]
@jschirrmacher Can you please check the builds? It seems to be a minor linting issue. FYI: Here are the logs incase you don't have access to it

```
@rocket.chat/meteor:lint: > @rocket.chat/meteor@6.0.0-develop eslint /home/runner/work/Rocket.Chat/Rocket.Chat/apps/meteor
@rocket.chat/meteor:lint: > eslint --ext .js,.jsx,.ts,.tsx .
@rocket.chat/meteor:lint: 
@rocket.chat/livechat:lint: 
@rocket.chat/livechat:lint: /home/runner/work/Rocket.Chat/Rocket.Chat/packages/livechat/src/store/Store.js
@rocket.chat/livechat:lint:   12:19  error  Replace `·return·store[name];·` with `⏎↹↹↹↹return·store[name];⏎↹↹↹`   prettier/prettier
@rocket.chat/livechat:lint:   13:24  error  Replace `·store[name]·=·val;·}` with `⏎↹↹↹↹store[name]·=·val;⏎↹↹↹},`  prettier/prettier
@rocket.chat/livechat:lint: 
@rocket.chat/livechat:lint: ✖ 2 problems (2 errors, 0 warnings)
@rocket.chat/livechat:lint:   2 errors and 0 warnings potentially fixable with the `--fix` option.
@rocket.chat/livechat:lint: 
```

File: packages/livechat/src/store/Store.js [hunk 1]
```diff
@@ -3,7 +3,19 @@ import mitt from 'mitt';
...
 
-const { localStorage, sessionStorage } = window;
+function getLocalStorage() {
+	try {
+		return window.localStorage;
+	} catch (_) {
+		const store = {};
+		return {
+			getItem(name) { return store[name]; },
+			setItem(name, val) { store[name] = val; }
```
Comment (jschirrmacher) [inline packages/livechat/src/store/Store.js:13]:
```suggestion
			getItem(name) {
				return store[name];
			},
			setItem(name, val) {
				store[name] = val;
			}
```

File: packages/livechat/src/store/Store.js [hunk 2]
```diff
@@ -3,7 +3,23 @@ import mitt from 'mitt';
...
+		return window.localStorage;
+	} catch (_) {
+		const store = {};
+		return {
+			getItem(name) {
+				return store[name];
+			},
+			setItem(name, val) {
+				store[name] = val;
+			}
```
Comment (murtaza98) [inline packages/livechat/src/store/Store.js:17]:
```suggestion
			},
```

**Review by KevLehman — APPROVED** [review]
Overall good, let's wait for FE to review it :)

Thanks for your contribution

**Review by MartinSchoeler — APPROVED** [review]

**Review by ggazzo — APPROVED** [review]

**Pull request merged**